### ISSUE TITLE
Steamworks.NET Wrapper that implements Unity's ISocialPlatform?

### ISSUE THREAD

**Issue by JashanChittesh** [rlabrecque/Steamworks.NET#245]
Apologies for using the issue tracker for something that's not a bug report and not even strictly a feature request - but this is the first place I looked for it and I also wasn't able to find anything about it in the Steam forum (I ran through 51 pages searching for "ISocial" on each page):

Has anyone tried using Steamworks.NET to build a wrapper that implements Unity's [ISocialPlatform](https://docs.unity3d.com/ScriptReference/SocialPlatforms.ISocialPlatform.html)?

I believe this would be really nice to be able to use the same API for Steamworks, [GameCenter](https://docs.unity3d.com/ScriptReference/SocialPlatforms.GameCenter.GameCenterPlatform.html), [Google Play Games](https://github.com/playgameservices/play-games-plugin-for-unity), and other social game platforms.

Obviously, ISocialPlatform only covers a small part of the Steamworks API but I still think this would already be quite useful with just that. For IAP, Unity also has an API, so that may be another thing that could be done.

**Comment by rlabrecque** [comment]
Not that I know of yet, that would be awesome though!

**Comment by creatio-ua** [comment]
Hi.

I’ve implemented some parts of the Social API using Steamworks.NET and also
a Unity IAP support for Steam in our last game. The code is not really
something I would be proud of, but it works (the hame is already
published). Specifically about the Social API, I have some kind of
implementation of: authorization, achievements and friends support (no
leaderboards etc.).

If that could be interesting to anybody, I would think about making it
public.

Best regards,
Alex Dovgodko

чт, 19 июля 2018 г. в 19:09, Riley Labrecque <notifications@github.com>:

**Comment by iamsed** [comment]
@creatio-ua sounds awesome if you could share it. I'd love to see the Unity IAP for Steam.

**Comment by creatio-ua** [comment]
@iamsed ok, I heard you sir! By the way, my IAP implementation uses a server side script (written on ASP.NET/C#) to verify payments and to call the game server to specify the game account which will receive the payment. 

So it won't work for off-line games out of the box.

**Comment by somethingSTRANGE** [comment]
@creatio-ua I'm interested in seeing your implementation as well. Thanks for sharing!

**Comment by mgrogin** [comment]
@creatio-ua - I'm interested in seeing your implementation!

**Comment by somethingSTRANGE** [comment]
@creatio-ua, are you still considering making your implementation public? I’m very interested in seeing how you handled it.

**Comment by creatio-ua** [comment]
Yes, just too short on time till the end of the month, to do anything about
it.

ср, 17 окт. 2018 г. в 21:32, somethingSTRANGE <notifications@github.com>:

**Comment by jcardussi** [comment]
Any work done on this? I am building the store now. I am hoping to wire it into Unity's IAP API.

**Comment by creatio-ua** [comment]
I've already started the work on preparing a version for publication on
git. Hope to finish it during the NY holidays.

вс, 23 дек. 2018 г. в 03:36, John C. Ardussi <notifications@github.com>:

**Comment by jcardussi** [comment]
That would be awesome! I will test it as soon as it is ready.

**Comment by creatio-ua** [comment]
Hi :)
I've finally did it. The documentation is not great yet, but the demo scene
should give some important info for the beginning.
https://github.com/mPowerGames/Steamworks-Platform

Would be great to have some feedback. I'll make a topic on the Unity forum
soon, to simplify the feedbacking.

P.S. I'm also going to deploy my Facebook and VKontakte implementation of
the ISocialPlatform if people will like the Steam version.

Best regards,
Alex.

ср, 26 дек. 2018 г. в 22:17, John C. Ardussi <notifications@github.com>:

**Comment by somethingSTRANGE** [comment]
Thanks, @creatio-ua! I'll take a look at your project.

**Comment by creatio-ua** [comment]
Welcome sir! :)
Will be waiting for your feedback.

вс, 13 янв. 2019 г. в 02:48, somethingSTRANGE <notifications@github.com>:
